axlearn/cloud/common/job.py

```diff
@@ -14,7 +14,17 @@
 
 
 class Job(Configurable):
-    """Base Job definition."""
+    """Base Job definition.
+
+    Job's main API method is `execute`, which sets up the environment according to `bundler`,
+    runs the specified `command`, and retries if necessary.
+
+    Subclasses of `Job` further specify the platform (e.g., TPUs on GCP) on which the job
+    should run.
+
+    The implementation of `execute` should be idempotent---invoking `execute` multiple times
```

Comment:
(As a side note, this is mostly necessary for jobs intended to be run by bastion.)